### ISSUE TITLE
Error when running tensorflow.python.ops.nn_impl.weighted_cross_entropy_with_logits_v2

### ISSUE THREAD

**Issue by nimashiri** [tensorflow/tensorflow#59339]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

No

### Source

binary

### Tensorflow Version

2.10.0

### Custom Code

Yes

### OS Platform and Distribution

Ubuntu 22.04

### Python version

3.9

### Current Behaviour?

```shell
2023-01-21 10:32:51.480512: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-01-21 10:32:51.592470: E tensorflow/stream_executor/cuda/cuda_blas.cc:2981] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2023-01-21 10:32:52.135367: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/
2023-01-21 10:32:52.135551: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/
2023-01-21 10:32:52.135560: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2023-01-21 10:32:52.699798: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-21 10:32:52.730300: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudnn.so.8'; dlerror: libcudnn.so.8: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/
2023-01-21 10:32:52.730316: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1934] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
2023-01-21 10:32:52.730557: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Error:<class 'tensorflow.python.framework.ops.EagerTensor'> returned a result with an error set
```
```


### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import os
import numpy as np
from tensorflow.python.ops import nn_impl
try:
  arg_0_tensor = tf.saturate_cast(tf.random.uniform([3, 3, 3], minval=0, maxval=2, dtype=tf.int64), dtype=tf.uint64)
  arg_0 = tf.identity(arg_0_tensor)
  arg_1_tensor = tf.cast(tf.random.uniform([3, 3, 3], minval=0, maxval=2, dtype=tf.int32), dtype=tf.bool)
  arg_1 = tf.identity(arg_1_tensor)
  arg_2 = -14
  arg_3 = True
  out = nn_impl.weighted_cross_entropy_with_logits_v2(arg_0,arg_1,arg_2,arg_3,)
except Exception as e:
  print("Error:"+str(e))
```
```


### Relevant log output

```shell
2023-01-21 10:32:51.480512: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-01-21 10:32:51.592470: E tensorflow/stream_executor/cuda/cuda_blas.cc:2981] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2023-01-21 10:32:52.135367: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/
2023-01-21 10:32:52.135551: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/
2023-01-21 10:32:52.135560: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2023-01-21 10:32:52.699798: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-21 10:32:52.730300: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudnn.so.8'; dlerror: libcudnn.so.8: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/
2023-01-21 10:32:52.730316: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1934] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
2023-01-21 10:32:52.730557: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Error:<class 'tensorflow.python.framework.ops.EagerTensor'> returned a result with an error set
```
```
</details>

**Comment by synandi** [comment]
Hi @nimashiri, please note that 


- The `labels` attribute accepts a Tensor of the same type and shape as logits, with values between 0 and 1 inclusive.
- The `logits` attribute accepts a Tensor of type float32 or float64, any real numbers.
- The `pos_weight` attribute accepts coefficient to use on the positive examples, typically a scalar but otherwise broadcastable to the shape of logits. Its value should be non-negative.

Kindly find the gist to the working code [here](https://colab.sandbox.google.com/gist/synandi/0d9497a58af70a81862c8af6e3375948/59339.ipynb). Thank you!

**Comment by mihaimaruseac** [comment]
@synandi vulnerabilities occur when invalid arguments are being sent to APIs. So your reply here is wrong.

@nimashiri this is not a vulnerability, you're receiving a user error. Please consult SECURITY.md before sending new reports as very few of them are really actionable (and they should have been submitted via private channels)
